src/translate.py

```diff
@@ -3206,6 +3432,8 @@ def output_regs_for_instr(
 ) -> List[Register]:
     def reg_at(index: int) -> List[Register]:
         reg = instr.args[index]
+        if isinstance(reg, AsmAddressMode):
+            return [reg.rhs]
```

Comment:
This needs a comment. It's about instrs_store_update, right? Maybe we should let that use a separate helper for clarity.

Actually, wait a minute... I don't see instrs_store_update mentioned in this function at all! Similarly instrs_load_update, and maybe others are missing too?

Reply:
I added `instrs_store_update`/`instrs_load_update`, but I need to revisit this to audit that everything matches `process_instr` and to comment/refaactor as you suggested.

Reply:
I don't know about a helper, because it'd still need to handle both the `Register` and ` AsmAddressMode` cases.
I did add comments + an `assert` to the branch.